### PR TITLE
Add TM2 offline validator utility pkg and service

### PR DESCRIPTION
**Pull request by rob05c**
This is dependent on changesets in PR #323, don't merge before it.

### CODE REVIEW

File: traffic_monitor_golang/traffic_monitor/tmcheck/tmcheck.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// package tmcheck contains utility functions for validating a Traffic Monitor is acting correctly.
```
Comment (dneuman64) [inline traffic_monitor_golang/traffic_monitor/tmcheck/tmcheck.go]: Looks like this file is missing the license header?

Comment (rob05c) [inline traffic_monitor_golang/traffic_monitor/tmcheck/tmcheck.go]: Fixed.

**Comment by dneuman64** [comment]
I kinda feel like the validator should be a stand-alone tool so we don't have to run goTM just to run the validator.  Thoughts?

**Comment by rob05c** [comment]
I'm not sure what you mean. It just hits endpoints, so you should be able to point it at a Java TM.

Or you mean, run it against a CRStates file? It should be easy enough to write an app to do that, using the `ValidateCRStates` function.

Or do you mean, have it automatically get monitors to check from Traffic Ops? I agree it should do that, it's just a bit more work to add that functionality.

**Comment by dneuman64** [comment]
Oh, I see now, validate-offline.go can be run stand-alone.  I was thinking you had to run goTM to get that functionality. I will take a deeper look this afternoon and merge if everything looks good.

**Pull request closed without merging**